tests/ui/autodiff_illegal.rs

```diff
@@ -0,0 +1,164 @@
+//@ needs-enzyme
+
+#![feature(autodiff)]
+//@ pretty-mode:expanded
+//@ pretty-compare-only
+//@ pp-exact:autodiff_illegal.pp
+
+// Test that invalid ad macros give nice errors and don't ICE.
+
+// We can't use Duplicated on scalars
+#[autodiff(df1, Reverse, Duplicated)]
+pub fn f1(x: f64) {
+//~^ ERROR     Duplicated can not be used for this type
+    unimplemented!()
+}
+
+// We can't use Dual on scalars
+#[autodiff(df2, Forward, Dual)]
+pub fn f2(x: f64) {
+    unimplemented!()
+}
```

Comment:
Problem: I think this is missing an ERROR annotation.

Reply:
Actually the opposite, it shouldn't have ended in this test file since it's not illegal, removed.